### PR TITLE
CI-83: Meme Selection

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-83](https://cloverpop-internship.atlassian.net/browse/CI-83)
On heroku should add var - `GIPHY_API_KEY`. This key  is available after registering on the [GIPHY](https://giphy.com/) website, please follow the instructions - https://docs.google.com/document/d/19VOqimOtENKUB0HqaPQ5-6iBx6YdJv90MBDwVQ8ZC_0/edit

[CI-83]: https://cloverpop-internship.atlassian.net/browse/CI-83?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rogergraves — APPROVED** [review]

**Review by SBorozenets — APPROVED** [review]

File: yarn.lock
Comment (SokalMet) [inline yarn.lock:1]: this file should not be in PR. Ask Vitalii Sh how to replace it from here.

File: app/javascript/components/Pages/Gifs/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  }, [term])
+
+  const GiphyLogo = () => <div className='d-flex justify-content-center align-items-center'>
+    <div className='fw-lighter mx-2 light-grey-text mt-1'>POWERED BY</div>
+    <div className='fw-bold h2 muted mt-2 mb-0'>GIPHY</div>
+  </div>
+
+  const Notice = () =>
+    <h1 className='text-white m-3'>We noticed that you didn't add the GIPHY api token for displaying gifs here. Please follow this&nbsp;
+    <Link to="https://docs.google.com/document/d/19VOqimOtENKUB0HqaPQ5-6iBx6YdJv90MBDwVQ8ZC_0/edit" target="_blank" rel="noopener noreferrer">
```
Comment (SokalMet) [inline app/javascript/components/Pages/Gifs/index.js:55]: such hardcoded vals should be in constant variables

File: app/javascript/components/__tests__/MemeSelection.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const mockService = { isLoading: false, error: null };
+    const data = {
+      emotion: {
+        category: "negative",
+        word: "anxious"
+      },
+      api_giphy_key: 'api_key',
+      response: {
+        attributes: {
+          gif_url: 'https://media.giphy.com/media/Ec8IY1BRRH3vrZfbcf/giphy.gif',
```
Comment (SokalMet) [inline app/javascript/components/__tests__/MemeSelection.test.js:27]: the same is here - hardcoded text. constant would be better to use

File: app/javascript/components/helpers/helpers.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+export const backHandling = () => {
+  window.history.back()
+}
+
+export const rangeFormat = (tp) => {
+  let start_date = new Date(tp.start_date)
+  let end_date = new Date(tp.end_date)
+  let month = end_date.toLocaleString('default', {month: 'long'}).slice(0,3)
```
Comment (SokalMet) [inline app/javascript/components/helpers/helpers.js:18]: those three variables should go as 'const', not 'let'

**Review by SokalMet — CHANGES_REQUESTED** [review]

File: yarn.lock
Comment (SerhiiBorozenets) [reply yarn.lock:1]: @SokalMet yarn.lock should be in the project, since its absence cause to the build error on heroku and does not pass the tests. 
https://dev.to/shrihankp/comment/1b213#:~:text=Jan%2030%20'21-,You%20should%20never%2C%20ever%20%22gitignore%22%20your%20lock%20files(,%22should%20commit%20this%20file%22.

Answer from Chat GPT:

If you're using Jest to run your tests and you're getting an error that says Cannot find module 'ramda/es/isEmpty', it's likely because Jest is unable to resolve the ramda module correctly.

To resolve this issue, you can try adding the following code to your Jest configuration:

javascript
Copy code
module.exports = {
  // other Jest configuration options
  moduleNameMapper: {
    "^ramda$": "ramda/src/index.js",
    "^ramda/es/(.*)$": "ramda/src/$1"
  }
};
This code maps the ramda module to the correct file path so that Jest can resolve it correctly. You can add this code to your Jest configuration file (usually jest.config.js or jest.config.ts), or add it directly to the jest object in your package.json file.

After making this change, Jest should be able to resolve the ramda module correctly and your tests should run without any issues.




Sergey Borozenets
yarn.lock should be in gitignore or not
It is generally recommended to include the yarn.lock file in your Git repository, rather than ignoring it in your .gitignore file.

The yarn.lock file is generated by Yarn and it includes specific version information for all of the packages and dependencies that your project uses. This ensures that all developers on your team are using the same package versions, which helps to prevent compatibility issues and ensures consistent behavior across different environments.

By including the yarn.lock file in your Git repository, you can ensure that all team members are working with the same dependencies and that any changes to the package versions are tracked and visible to everyone.

Ignoring the yarn.lock file in your .gitignore file can lead to inconsistencies and issues between different environments, as developers may end up using different package versions. It can also make it more difficult to reproduce issues or debug problems, since you may not have access to the exact package versions that were used when the issue occurred.

In summary, it's generally a good idea to include the yarn.lock file in your Git repository, rather than ignoring it in your .gitignore file.

Comment (SokalMet) [reply yarn.lock:1]: @SerhiiBorozenets  basically heroku is running yarn to install your dependencies, and yarn.lock file is generated after. So this file will **always be in your project**. With my experience - better to remove this file from PR`s. But if you have decided in your team to leave this file in PR`s, so go ahead.
